src/TableFunctions/TableFunctionObjectStorage.cpp

```diff
@@ -165,9 +167,17 @@ StoragePtr TableFunctionObjectStorage<Definition, Configuration>::executeImpl(
         return storage;
     }
 
+    if (insert_query)
+    {
+        if (const auto * insert_query_class = insert_query->as<ASTInsertQuery>())
+        {
+            configuration->updatePartitionStrategy(insert_query_class->partition_by, columns, context);
+        }
+    }
```

Comment:
Should be exception LOGICAL_ERROR if `insert_query` is not `ASTInsertQuery`? Or it can be legal?